jsk_tools/src/sanity_diagnostics.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python
+
+# https://github.com/start-jsk/jsk_apc/blob/master/jsk_arc2017_baxter/node_scripts/sanity_check_for_pick.py # NOQA
+
+from diagnostic_msgs.msg import DiagnosticArray, DiagnosticStatus
+from jsk_tools.sanity_lib import checkNodeState, checkTopicIsPublished
+import os
+import rospy
+import yaml
+
+
+class SanityDiagnostics(object):
+    """
+    This node publishes essential robot topic and node status to /diagnostics.
+
+    The yaml file is like the following:
+    topics:
+      - /kinect_head/rgb/image_raw
+      - /tf
+    nodes:
+      - /kinect_head/kinect_head_nodelet_manager
+      - /respeaker_node
+    """
+    def __init__(self):
+        yaml_path = rospy.get_param(
+            '~sanity_targets', "/var/lib/robot/sanity_targets.yaml")
+        if os.path.exists(yaml_path):
+            with open(yaml_path) as yaml_f:
+                sanity_targets = yaml.load(yaml_f)
+                self.topics = sanity_targets['topics']
+                self.nodes = sanity_targets['nodes']
+                rospy.loginfo(
+                    "{} is loaded as sanity check target file.".format(
+                        yaml_path))
+                rospy.loginfo(sanity_targets)
+            self.pub = rospy.Publisher(
+                '/diagnostics', DiagnosticArray, queue_size=10)
+            self.timer = rospy.Timer(rospy.Duration(1), self.check_sanity)
+
+    def check_sanity(self, event):
+        self.checkNode()
+        self.checkTopic()
+
+    def checkNode(self):
+        diag_array_msg = DiagnosticArray()
+        diag_array_msg.header.stamp = rospy.Time.now()
+        status_array = []
+        for node in self.nodes:
+            node_state = checkNodeState(node, needed=True)
+            state_msg = DiagnosticStatus()
+            state_msg.name = node
+            if node_state:
+                state_msg.level = DiagnosticStatus.OK
+                state_msg.message = 'Node is alive'
+            else:
+                state_msg.level = DiagnosticStatus.ERROR
+                state_msg.message = 'Node is dead'
+            status_array.append(state_msg)
+        diag_array_msg.status = status_array
+        self.pub.publish(diag_array_msg)
+
+    def checkTopic(self):
```

Comment:
I prefer `check_topic`

Reply:
Thank you!

I fixed in
[`2dcf164` (#1727)](https://github.com/jsk-ros-pkg/jsk_common/pull/1727/commits/2dcf164b2e5c72196f3e720d13a96f12ed2afd59)